### PR TITLE
Install and use python 3.6.3 for circleci integration tests

### PR DESCRIPTION
**Pull request by jchengsfx**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -169,7 +170,7 @@ jobs:
           export AGENT_BIN=$(pwd)/bundle/bin/signalfx-agent
           export BUNDLE_DIR=$(pwd)/bundle
           export TEST_SERVICES_DIR=$(pwd)/test-services
-          sudo -E pytest \
+          sudo -E $PYENV_ROOT/shims/pytest \
```
Comment (keitwb) [inline .circleci/config.yml]: Why do we need this??

Comment (jchengsfx) [reply .circleci/config.yml]: `$PYENV_ROOT/shims/` is auto added to PATH for the circleci user by pyenv, but sudo protects PATH.

File: .circleci/config.yml [hunk 2]
```diff
@@ -250,6 +252,19 @@ jobs:
       - store_artifacts:
           path: ~/testresults/k8s_integration_tests.html
 
+  int_test_lint:
```
Comment (jchengsfx) [inline .circleci/config.yml]: @keitwb Did you want to add this job to the workflow?

Comment (keitwb) [reply .circleci/config.yml]: Yes

**Pull request merged**